File: DS Implementations (in C)/Linked Lists/LL.c
#include <stdio.h>
#include <stdlib.h>
struct node
{
    int data;
    struct node* next;
};

void print_LL(struct node*);

int main()
{
    struct node* head, *ptr, * save;
    head = NULL;

    int input;
    char choice = 'Y';

    while(choice == 'Y' || choice == 'y')
    {
        printf("Enter Element Number: \n");
        scanf("%d", &input);

        ptr = (struct node*)malloc(sizeof(struct node));
        ptr -> data = input;

        if(head == NULL)
        {
            head = ptr;
            save = ptr;
        }
        else
        {
            save -> next = ptr;
            save = ptr;
        }

        printf("Would You Like to add another element?\n");
        scanf(" %c", &choice);
    }

    print_LL(head);

    return 0;
}

void print_LL(struct node* head)
{
    struct node* ptr = head;

    while(ptr != NULL)
    {
        printf("%d --> ", ptr->data);
        ptr = ptr->next;
    }
}

File: DS Implementations (in C)/Linked Lists/LL.py
# Python3 implementation of LinkedList

class Node:
    def __init__(self, value = 0, next = None): # Default value of a node is 0
        self.data = value
        self.next = next

    def Print(self, head):
        ptr = head

        while(ptr != None):
            print(f"{ptr.data} --> ", end = "")
            ptr = ptr.next

        print()

def main():
    head_value = int(input("Enter head value: "))
    head = Node(head_value)

    while(True):
        choice = int(input("Enter 1 to Enter new node, 2 to print and 3 for exit: "))
        if(choice == 1):
            temp = int(input("Enter Node Value: "))
            ptr = head
            while(ptr.next != None):
                ptr = ptr.next

            new_node = Node(temp)

            ptr.next = new_node

        elif(choice == 2):
            head.Print(head)

        else:
            break

            


if __name__ == "__main__":
    main()